### PR TITLE
User Authentication(Login Page)

### PR DESCRIPTION
**Pull request by Sudhindra3000**
### Features:

- Created login page with basic functionality
- Show Login page or Home page on app start, based on auth state
- Navigation using named routes
- Show snackbar when user logs in or logs out
- Store User data and sessionId in UserLoggedIn state
- Persist state in UserBloc on all platforms using [HydratedBloc](https://pub.dev/packages/hydrated_bloc)
- Added Documentation for some classes and functions

### Todo:

- [ ] Encrypt Session Id
- [ ] Use proper dependency injection to access sessionId accros the app
- [ ] Sign up pages
- [ ] Better and simpler routing strategy (RouteGenerator pattern is fine, but too much boilerplate is needed)
- [ ] Form validation in Login Page

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -11,25 +11,28 @@ dependencies:
...
   flutter_test:
     sdk: flutter
   flutter_lints: ^1.0.0
-# Package for adding app icon uncomment these lines whenever we need to update the app icon
-#   flutter_launcher_icons: "^0.9.2"
-
-# flutter_icons:
-#   android: "launcher_icon"
-#   ios: true
-#   image_path: "assets/icon/dalal_street_icon.png"
```
Comment (dhruv1294) [inline pubspec.yaml:32]: removed the app icon??

**Review by satyamurti — DISMISSED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (satyamurti) [reply pubspec.yaml:32]: Add these lines back , we need this package in future for updating app icon.

Comment (dhruv1294) [reply pubspec.yaml:32]: ++

File: lib/main.dart [hunk 2]
```diff
@@ -1,33 +1,81 @@
...
+      title: 'Dalal Street 2021',
+      navigatorKey: _navigatorKey,
+      debugShowCheckedModeBanner: false,
+      theme: ThemeData(
+          textTheme: GoogleFonts.rubikTextTheme(Theme.of(context).textTheme)),
+      // Show snackbar and navigate to Home or Login page whenever UserState changes
+      builder: (context, child) => BlocListener<UserBloc, UserState>(
+        listener: (context, state) {
+          if (state is UserLoggedIn) {
+            print('user logged in');
```
[the remark this answers is not in the document]
Comment (Sudhindra3000) [reply lib/main.dart:57]: Bcuz sometimes state management can get tricky as we're just in early stages

File: pubspec.yaml (hunk 1 above)
Comment (Sudhindra3000) [reply pubspec.yaml:32]: Ok I'll add it back

File: lib/main.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhruv1294) [reply lib/main.dart:57]: i also think for dev stages we should keep logs

**Review by dhruv1294 — APPROVED** [review]

File: lib/main.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (satyamurti) [reply lib/main.dart:57]: Cools👍

**Review by satyamurti — APPROVED** [review]

**Pull request merged**